content/blog/blog-2020-06-29-kuttl-v0.5.0.md

```diff
@@ -0,0 +1,139 @@
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-web
+status:
+  phase: Running
+```
+
+Running a `kubectl kuttl assert --timeout 5 01-assert.yaml` might provide an successful response such as:
+`"01-assert.yaml" in "default" namespace is valid`
+or it may provide a verbose output which ends with the details such as:
+
+`resource Pod:default/static-web: .status.phase: value mismatch, expected: Running != actual: Pending`
+
+::: warning
+These examples expect that you have a running kubernetes cluster and that kuttl can connect to it.
+:::
+
+The `errors` command provides the absolute affect, it asserts that the value doe NOT exist in the cluster. Example:
+```
+kubectl kuttl errors --timeout 5 01-assert.yaml
+error assert is valid
+```
+
+::: tip
+These commands work with URLs as well.  Example: `kubectl kuttl errors https://gist.githubusercontent.com/kensipe/ccf11977a1f9816865beae1a6e4e7883/raw/df0f0432387486410b5a3da20b5d45fead9e91b0/assert.yaml --timeout 1
+error assert is valid`
+:::
+
+
+## Namespace Control
+
+Previous to v0.5.0 KUTTL would always create a test namespace.  This works great in environments where you have full control of your cluster, however many orgs were unable to use kuttl for this reason allow.  It is now possible to pass a namespace to kuttl to use.  When a namespace is provided, kuttl will use that namespace if it exists and attempts to create if it doesn't.  The justification for creation is the test suite would have failed if the namespace didn't exist, if the kuttl user isn't authorized to create a namespace it will also fail. 
+
+:::warning
+When a namespace is NOT provided to KUTTL, it will create namespace for each test in the testsuite providing some test isolation.  When a namespace is provided, that namespace is used for all tests in the test suite.
+:::
+
+:::warning 
+KUTTL deletes namespaces which it creates.  If the namespace was NOT created by KUTTL it does not delete it.
+:::
+
+An example run is: `kubectl kuttl test ~/projects/playground/e2e/ -n default` which will runs under the default namespace.
+
+## Command Expansion
+
+Prior to v0.5.0, KUTTL provided command expanstion of ENV VARs in commands for `TestStep`.  This allowed for a command that used `$NAMESPACE` in the command.  This lead to requests for use in other areas... In particular newly added features for `TestStep` using `apply`, `errors` and `assert`.  The following is now possible:
+
+```
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+timeout: 30
+apply:
+  - ${TEST_FOO}/test.yaml
+assert:
+  - $TEST_FOO/test.yaml
+```
+
+This does require that `$TEST_FOO` is defined.
+
+:::warning
+It is worth noting regarding the expansion of `commands` that it doesn't have full shell support and is limited.  There is a new command option `script` which runs the command in a shell, which provides more shell ability.  Example:
+
+```
+commands:
+    - command: ls -l
+    - script: for i in {1..5}; do echo $NAMESPACE; done
```

Comment:
A multiline example would be better

Reply:
I'm not sure how it would make it better... this blog post was to introduce the concept ahead of real documentation.  we can be more exhaustive in the docs... right?